addons/website_event/controllers/main.py

```diff
@@ -305,14 +305,20 @@ def _create_attendees_from_registration_post(self, event, registration_data):
         if visitor_values:
             visitor_sudo.write(visitor_values)
 
-        return request.env['event.registration'].sudo().create(registrations_to_create)
+        return request.env['event.registration'].sudo().create(registrations_to_create), visitor_sudo
 
     @http.route(['''/event/<model("event.event"):event>/registration/confirm'''], type='http', auth="public", methods=['POST'], website=True)
     def registration_confirm(self, event, **post):
         registrations = self._process_attendees_form(event, post)
-        attendees_sudo = self._create_attendees_from_registration_post(event, registrations)
+        attendees_sudo, visitor_sudo = self._create_attendees_from_registration_post(event, registrations)
 
-        return request.redirect(('/event/%s/registration/success?' % event.id) + werkzeug.urls.url_encode({'registration_ids': ",".join([str(id) for id in attendees_sudo.ids])}))
+        redirect = request.redirect(('/event/%s/registration/success?' % event.id) + werkzeug.urls.url_encode({'registration_ids': ",".join([str(id) for id in attendees_sudo.ids])}))
+
+        if request.httprequest.cookies.get('visitor_uuid', '') != visitor_sudo.access_token:
```

Comment:
it would be useful to add a comment line about when this can happen (when the administrator shoots himself in the foot by disabling "Track Visitor"), since this seems like a very unlikely case to happen so it's not too clear why the code is needed just by reading the code

Reply:
Thank you! I will add a comment for it :pray: